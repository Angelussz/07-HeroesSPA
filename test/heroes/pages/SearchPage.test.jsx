import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter, useNavigate } from "react-router-dom";
import { SearchPage } from "../../../src/heroes";
const mockedUseNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUseNavigate,
}));
describe("Pruebas en <SearchPage />", () => {
    beforeEach(()=>jest.clearAllMocks())
  test("debe mostrar correctamente los valores por defecto", () => {
    const { container } = render(
      <MemoryRouter>
        <SearchPage />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
  test("debe mostrar a batman y el input con valor de queryString", () => {
    render(
      <MemoryRouter initialEntries={["search?q=batman"]}>
        <SearchPage />
      </MemoryRouter>
    );
    const input = screen.getByRole("textbox");
    expect(input.value).toBe("batman");
    const img = screen.getByRole("img");
    expect(img.src).toContain("/assets/heroes/dc-batman.jpg");
    // console.log(screen.getByText("Search Hero"))
    expect(screen.getByText("Search Hero").style.display).toContain("none");
  });
  test("debe de mostrar un error si no se encuentra el hero(batman123)", () => {
    render(
      <MemoryRouter initialEntries={["search?q=batman123"]}>
        <SearchPage />
      </MemoryRouter>
    );
    expect(screen.getByText("Search Hero").style.display).toContain("none");
    expect(
      screen.getByText("No hay resultados con el heroe").style.display
    ).not.toContain("none");
  });
  test("debe de llamar el navigate a la pantalla nueva", () => {
    const inputValue = "superman"
    render(
      <MemoryRouter initialEntries={["/search"]}>
        <SearchPage />
      </MemoryRouter>
    );
    const input = screen.getByRole("textbox");
    fireEvent.change(input, {
      target: {
        name: "searchText",
        value: inputValue
      },
    });
    const formHero = screen.getByLabelText("formHero");
    fireEvent.submit(formHero);
    expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${inputValue}`);
  });
});
