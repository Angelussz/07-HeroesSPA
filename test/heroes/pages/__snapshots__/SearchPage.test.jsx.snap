// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe mostrar correctamente los valores por defecto 1`] = `
<div>
  <h1>
    Busqua al heroe que quieras!
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Busqueda
      </h4>
      <hr />
      <form
        aria-label="formHero"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Busca Heroe"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Buscar
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Resultados
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search Hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hay resultados con el heroe 
        <b />
         
      </div>
    </div>
  </div>
</div>
`;
